cfn-resources/global-cluster-config/cmd/resource/resource.go

```diff
@@ -0,0 +1,376 @@
+		_, _ = logger.Debugf("error creating MongoDB Global Cluster Configuration: %s", err)
+		var target *mongodbatlas.ErrorResponse
+		if errors.As(err, &target) && target.ErrorCode != "DUPLICATE_MANAGED_NAMESPACE" {
+			return handler.ProgressEvent{
+				OperationStatus:  handler.Failed,
+				Message:          err.Error(),
+				HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest,
+			}, nil
+		}
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          err.Error(),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeServiceInternalError,
+		}, nil
+	}
+
+	// add zone mappings
+	customZoneMappings := modelToCustomZoneMappings(currentModel.CustomZoneMappings)
+
+	// API call to create
+	clusterDetail, _, err := client.GlobalClusters.AddCustomZoneMappings(context.Background(), projectID, clusterName, &mongodbatlas.CustomZoneMappingsRequest{
+		CustomZoneMappings: customZoneMappings,
+	})
+	if err != nil {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          err.Error(),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeServiceInternalError,
+		}, nil
+	}
+	responseData, _ := json.Marshal(clusterDetail)
+	_, _ = logger.Debugf("Response Object: %s", responseData)
+
+	event := handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Completed",
+		ResourceModel:   currentModel,
+	}
+	_, _ = logger.Debugf("Create() return event:%+v", event)
+	return event, nil
+}
+
+func Read(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	setup() // logger setup
+	_, _ = logger.Debugf("Read snapshot for Request() currentModel:%+v", currentModel)
+
+	// Validate required fields in the request
+	if modelValidation := validateModel(RequiredFields, currentModel); modelValidation != nil {
+		return *modelValidation, errors.New("required field not found")
+	}
+
+	// Create MongoDb Atlas Client using keys
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey, *currentModel.ApiKeys.PrivateKey)
+	if err != nil {
+		_, _ = logger.Warnf(constants.ErrorCreateMongoClient, err)
+		return progressevents.GetFailedEventByCode(fmt.Sprintf("Failed to Create Client : %s", err.Error()),
+			cloudformation.HandlerErrorCodeInvalidRequest), err
+	}
+
+	// Check if  already exist
+	if !isExist(*client, currentModel) {
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          "Resource Not Found",
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
+	}
+	// method call to read configuration
+	config, _, err := ReadConfig(*client, currentModel)
+	if err != nil {
+		_, _ = logger.Debugf("error reading MongoDB Global Cluster Configuration (%s): %v", *currentModel.ClusterName, err)
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          err.Error(),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeNotFound}, nil
```

Comment:
if the ReadConfig is returning a progress event, why we are returning a new one? and also all the errors coming from the ReadConfig we are returning a HandlerErrorCodeNotFound?